src/components/item/sharing/ConfirmMembership.tsx

```diff
@@ -0,0 +1,92 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+
+import { PermissionLevel } from '@graasp/sdk';
+import { Button } from '@graasp/ui';
+
+import { useCurrentUserContext } from '@/components/context/CurrentUserContext';
+
+import { useBuilderTranslation } from '../../../config/i18n';
+import { mutations } from '../../../config/queryClient';
+import { CONFIRM_MEMBERSHIP_DELETE_BUTTON_ID } from '../../../config/selectors';
+import { BUILDER } from '../../../langs/constants';
+import CancelButton from '../../common/CancelButton';
+
+const labelId = 'alert-dialog-title';
+const descriptionId = 'alert-dialog-description';
+
+type Props = {
+  open?: boolean;
+  handleClose: () => void;
+  item: any;
+  memberToDelete: any;
+  isOneAdmin: boolean;
+};
+
+const DeleteItemDialog = ({
+  item,
+  open = false,
+  handleClose,
+  memberToDelete,
+  isOneAdmin = false,
+}: Props): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+  const { data: member } = useCurrentUserContext();
+
+  const { mutate: deleteItemMembership } = mutations.useDeleteItemMembership();
+
+  const onDelete = () => {
+    deleteItemMembership({ id: memberToDelete.instanceId, itemId: item.id });
+    handleClose();
+  };
+
+  let dialogText = '';
+  // incase of deleting the only admin
+  if (isOneAdmin && memberToDelete?.permission === PermissionLevel?.Admin) {
+    dialogText = translateBuilder(BUILDER?.DELETE_ONLY_ADMIN_ALERT_MESSAGE);
+  } else if (member?.id === memberToDelete?.memberInstance) {
```

Comment:
I think it's strange to check for member existence here. Probably we want to have a check higher `if(!member)` and show an error message? Most probably you won't ever the error in real cases.

Reply:
thanks a lot @pyphilia  for all your comments, 
Regarding this point the first condition was created to check if we are going to delete the only admin, So to check first if this item has only one admin and then to check if we're going to delete this one admin!
Please correct me If I'm wrong

Reply:
Yes! But my point is about `?.`. Probably you should have 

```
if(memberToDelete && member) {
if (isOneAdmin && memberToDelete.permission === PermissionLevel.Admin) {
    dialogText = translateBuilder(BUILDER?.DELETE_ONLY_ADMIN_ALERT_MESSAGE);
  } else if (member.id === memberToDelete.memberInstance) {
...
}
```